rules/jsx-import.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+const arrayToSentence = require('array-to-sentence');
+const astUtils = require('eslint-ast-utils');
+const getDocsUrl = require('./utils/get-docs-url');
+
+const pragmas = {
+	h: ['dom-chef', 'preact'],
+	React: ['react'],
+	preact: ['preact']
+};
+
+const pragmaAPIs = {
+	React: ['cloneElement']
+};
+
+const getSuggestedPackages = modules => arrayToSentence(modules.map(x => `\`${x}\``), {lastSeparator: ' or '});
+
+const hasPackage = (node, packages) => node.init ?
+	packages.some(x => x === astUtils.getRequireSource(node.init)) :
+	packages.some(x => x === node.parent.source.value);
+
+const isImportDeclaration = node => node.init ?
+	astUtils.isStaticRequire(node.init) :
+	node.parent.type === 'ImportDeclaration';
+
+const getVariable = (name, variables) => variables.find(x => x.name === name);
+
+// https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/util/variable.js#L27-L52
+const getVariablesInScope = context => {
+	let scope = context.getScope();
+	let {variables} = scope;
+
+	while (scope.type !== 'global') {
+		scope = scope.upper;
+		variables = variables.concat(scope.variables);
+	}
+
+	if (scope.childScopes.length > 0) {
+		variables = scope.childScopes[0].variables.concat(variables);
+
+		if (scope.childScopes[0].childScopes.length > 0) {
+			variables = scope.childScopes[0].childScopes[0].variables.concat(variables);
```

Comment:
Would be nice to know if and why is this needed. Already seen it copy-pasted here https://github.com/sindresorhus/eslint-plugin-unicorn/pull/224#discussion_r251252397

Reply:
Yeah, and would be good to have some separate tests for util methods.